[BREAKING] Enable ABI coder v2 by default. (#10336)
ethereum/solidity: docs/080-breaking-changes.rst

  Checks for overflow are very common, so we made them the default to increase readability of code,
  even if it comes at a slight increase of gas costs.

* ABI coder v2 is activated by default.

  You can choose to use the old behaviour using ``pragma abicoder v1;``.

>   The pragma ``pragma experimental ABIEncoderV2;`` is still valid (although it does not change anything),

Reviewer: ```suggestion
  The pragma ``pragma experimental ABIEncoderV2;`` is still valid, but it is deprecated and has no effect. If you want to...
```
?

>   The pragma ``pragma experimental ABIEncoderV2;`` is still valid, but it is deprecated and has no effect. If you want to...
>   but if you want to be explicit, please use ``pragma abicoder v2;`` instead.

Reviewer: ```suggestion
  The pragma ``pragma experimental ABIEncoderV2;`` is still valid, but it is deprecated and has no effect.
  If you want to be explicit, please use ``pragma abicoder v2;`` instead.
```

This time a proper two-line suggestion without ellipses, s.t. it's fine to directly apply ;-). Sorry for not doing that before.

Author: sorry, should have paid more attention.

>   Note that ABI coder v2 supports more types than v1 and performs more sanity checks on the inputs.
>   The latter makes some function calls more expensive and it can also make contract calls
>   revert that did not revert with ABI coder v1 when they contain data that does not conform to the
>   parameter types.

Reviewer: ```suggestion
  Note that ABI coder v2 supports more types than v1 and performs more sanity checks on the inputs.
  ABI coder v2 makes some function calls more expensive and it can also make contract calls
  revert that did not revert with ABI coder v1 when they contain data that does not conform to the
  parameter types.
```

Or ``The former``, right?
I also had a larger comment because I think this paragraph may make it sound a bit like a bad thing (less calls work and those who do work are more expensive :-D)... but github swallowed that because it was at the moment of rebase? That's sad...
